app/assets/javascripts/discourse/models/user-drafts-stream.js.es6

```diff
@@ -0,0 +1,99 @@
+import { ajax } from "discourse/lib/ajax";
+import { url } from "discourse/lib/computed";
+import RestModel from "discourse/models/rest";
+import UserDraft from "discourse/models/user-draft";
+import { emojiUnescape } from "discourse/lib/text";
+import computed from "ember-addons/ember-computed-decorators";
+
+import {
+  NEW_TOPIC_KEY,
+  NEW_PRIVATE_MESSAGE_KEY
+} from "discourse/models/composer";
+
+export default RestModel.extend({
+  loaded: false,
+
+  init() {
+    this._super();
+    this.setProperties({
+      itemsLoaded: 0,
+      content: [],
+      lastLoadedUrl: null
+    });
+  },
+
+  baseUrl: url(
+    "itemsLoaded",
+    "user.username_lower",
+    "/drafts.json?offset=%@&username=%@"
+  ),
+
+  load() {
+    this.setProperties({
+      itemsLoaded: 0,
+      content: [],
+      lastLoadedUrl: null
+    });
+    return this.findItems();
+  },
+
+  @computed("content.length", "loaded")
+  noContent(contentLength, loaded) {
+    return loaded && contentLength === 0;
+  },
+
+  remove(draft) {
+    let content = this.get("content").filter(
+      item => item.sequence !== draft.sequence
+    );
+    this.setProperties({ content, itemsLoaded: content.length });
+  },
+
+  findItems() {
+    let findUrl = this.get("baseUrl");
+
+    const lastLoadedUrl = this.get("lastLoadedUrl");
+    if (lastLoadedUrl === findUrl) {
+      return Ember.RSVP.resolve();
+    }
+
+    if (this.get("loading")) {
+      return Ember.RSVP.resolve();
+    }
+
+    this.set("loading", true);
+
+    return ajax(findUrl, { cache: "false" })
```

Comment:
We need to handle errors here as well.

Reply:
I tried adding an error handler here, but I couldn't figure out how to test it, how to trigger it. Plus, similar models ([user-stream](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/models/user-stream.js.es6), [user-posts-stream](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/models/user-posts-stream.js.es6)) don't have error handling for similar calls.